core/modules/field/field.block.inc

```diff
@@ -80,9 +80,34 @@ class FieldBlock extends Block {
     list($entity_type, $field_name) = explode('-', $this->childDelta);
     $entity = $this->contexts[$entity_type]->data;
 
-    // Do not render if the entity is not found on this layout.
-    if (empty($entity)) {
-      return NULL;
+    // Do not proceed if the entity is not an EntityInterface object
+    if (!is_a($entity, 'EntityInterface')) {
```

Comment:
Why not even `    if ($entity_type == 'user' && !is_a($entity, 'EntityInterface') && isset($entity->uid)) {` ?

Reply:
This is somewhat moot now.